internetarchive/openlibrary
# Add patron action icons in book page sidebar

Closes #6557


Builds upon #6469

Replaces giant button that opens the share modal with a small icon with text.  Creates similar icons for reviews and notes modals, and places all three icons beneath the star ratings component.

Removes review icon link from star ratings component.

### Technical

A few things:

Modal link templates have been modified such that the markup for the triggering link is passed as an argument (as opposed to passing the anchor's content).  This should give us more flexibility when using these templates in the future.

`data-context` in modal link components is now attached to the modal content (instead of the triggering `<a>` element, which no longer exists in modal templates).

A guard has been added IOT prevent the review component from rendering when the patron is not logged in.  The review component requires a logged in user to operate properly.

Click listeners for the share modal link have been moved to `models/index.js`.  This file contains the client-side functionality for our notes and reviews modal links.  Modal triggering function in this file are now more generalized, and can be used by components that do not include `data-context` attributes.

If you see the old share image while testing, you may need to clear your browser's image cache.  This was an issue I encountered in FF.

New Google Analytics events have been created for the icon links:

| Trigger | Category | Action | 
|---|---|---|
| Share icon link click | ModalLinkClick | ShareIcon |
| Notes icon link click | ModalLinkClick | NoteIcon |
| Review icon link click | ModalLinkClick | ReviewIcon |

### Testing

While logged in:
1. Ensure that each new modal icon link opens the correct corresponding modal on a book page.
2. Ensure that the social modal links are correct, and that they link to the correct pages.
3. Ensure that the embed HTML link opens a dialog with the expected HTML.
4. Click the textual "Add your community review" link, found at the bottom of the review stats component.  Ensure that a review modal opens and works as expected.
5. Navigate to the "My Reviews" view in your account page.  Ensure that the review modals work properly, and that the reviews in "My Reviews" update when a change is made and the modal is closed.

While logged out:
1. Navigate back to a book page.
2. Click on each of the new modal icon links.  Ensure that the notes and reviews modal links take you to our login page, and that the correct redirect parameter is present in the URL.
3. Ensure that the share modal icon link opens the share modal, as it does when you are logged in.

### Screenshot

![modal_icons](https://user-images.githubusercontent.com/28732543/169617537-3fb3be5b-afc1-4043-b453-0eb2e14942ba.png)

### Stakeholders
